app/app/vhosts/edit/route.js

```diff
@@ -1,55 +1,99 @@
 import Ember from 'ember';
-import DS from 'ember-data';
 
 export default Ember.Route.extend({
-  title: function(){
+  title() {
     let app = this.modelFor('app');
 
-    return `Edit ${this.currentModel.get('virtualDomain')} - ${app.get('handle')}`;
+    return `Edit ${this.currentModel.vhost.get('virtualDomain')} - ${app.get('handle')}`;
   },
 
-  afterModel: function(model){
-    return model.get('service');
+  model(params) {
+    let app = this.modelFor('app');
+    let stack = app.get('stack');
+
+    return Ember.RSVP.hash({
+      vhost: this.store.find('vhost', params.vhost_id),
+      certificates: stack.get('certificates'),
+      stack: stack
+    });
+  },
+
+  afterModel(model) {
+    let vhost = model.vhost;
+
+    return Ember.RSVP.hash({
+      service: vhost.get('service'),
+      certificate: vhost.get('certificate')
+    });
+  },
+
+  setupController(controller, model) {
+    let vhost = model.vhost;
+    let certificates = model.certificates;
+
+    controller.set('model', vhost);
+    controller.set('certificates', model.certificates);
+    controller.set('vhostCertificate', certificates.objectAt(0));
+    controller.set('showAddNewCertificate', model.certificates.get('length') === 0);
```

Comment:
Seems like this should be a CP on the controller.  Perhaps, `Ember.computed.eq`?